src/backend/keys/cswordkey.cpp

```diff
@@ -59,8 +59,21 @@ QString CSwordKey::renderedText(const CSwordKey::TextRenderType mode) {
 
     auto & m = m_module->module();
     sword::VerseKey * vk_mod = dynamic_cast<sword::VerseKey *>(m.getKey());
-    if (vk_mod)
+    if (vk_mod) {
+        CSwordVerseKey * verseKey = static_cast<CSwordVerseKey *>(this);
```

Comment:
@jaakristioja still not sure if its good. This change would be done in override method.

Reply:
Regardless of whether something like this needs to be virtual or not, I'm not sure whether this is a correct place for this code either, because `CSwordKey::renderedText()` has multiple callsites, all of which might not need this logic.

Reply:
Yes, i moved it here just because of this, so whenever someone do `CSwordVerseKey key(bibleModule); key.setKey("Gen.1.1-2");` it will render range of keys, that looks correct behavior. It is looks correct for cross reference rendering, footnote rendering, text main view (it may display range of keys if there is versification mapping that caused key expanding). 

Let's wait before you consider this.